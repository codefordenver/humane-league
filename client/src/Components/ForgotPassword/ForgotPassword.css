.ForgotPassword {
  display: grid;
  grid-template-columns: 1fr 1fr;
  width: 600px;
  margin: 0px auto;
  grid-gap: 10px; }
  .ForgotPassword h1 {
    grid-column: span 2;
    text-align: center;
    margin: 0px; }
  .ForgotPassword a {
    grid-column: span 2;
    text-align: center; }
  .ForgotPassword label {
    display: flex;
    align-items: center; }
  .ForgotPassword button {
    grid-column: span 2;
    height: 70px;
    display: flex;
    width: 250px;
    align-items: center;
    justify-content: center;
    border-radius: 2px;
    border: none;
    background: #1CB841;
    color: white;
    cursor: pointer;
    margin: 0px auto;
    transition: 0.3s;
    font-size: 20px;
    padding: 10px; }
    .ForgotPassword button:hover {
      color: white;
      background-color: #323232; }
    .ForgotPassword button:disabled {
      background: #989696;
      color: black; }
  .ForgotPassword input {
    height: 50px;
    font-size: 24px;
    padding-left: 10px;
    outline: none;
    grid-column: span 2; }
  .ForgotPassword span {
    grid-column: span 2;
    text-align: center;
    font-size: 16px;
    min-height: 16px; }
