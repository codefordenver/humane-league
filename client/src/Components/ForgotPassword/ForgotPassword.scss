.ForgotPassword {
  display: grid;
  grid-template-columns: 1fr 1fr;
  width: 600px;
  margin: 0px auto;
  grid-gap: 10px;

  h1 {
    grid-column: span 2;
    text-align: center;
    margin: 0px;
  }

  a {
    grid-column: span 2;
    text-align: center;
  }

  label {
    display: flex;
    align-items: center;
  }

  button {
    grid-column: span 2;
    height: 70px;
    display: flex;
    width: 250px;
    align-items: center;
    justify-content: center;
    border-radius: 2px;
    border: none;
    background: #1CB841;
    color: white;
    cursor: pointer;
    margin: 0px auto;
    transition: 0.3s;
    font-size: 20px;
    padding: 10px;
    &:hover {
      color: white;
      background-color: rgb(50, 50, 50);
    }
    &:disabled {
      background: rgb(152, 150, 150);
      color: black;
    }
  }

  input {
    height: 50px;
    font-size: 24px;
    padding-left: 10px;
    outline: none;
    grid-column: span 2;
  }

  span {
    grid-column: span 2;
    text-align: center;
    font-size: 16px;
    min-height: 16px;
  }

}