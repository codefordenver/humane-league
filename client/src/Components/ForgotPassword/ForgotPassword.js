import React, { Component } from 'react';
import './ForgotPassword.css';
import firebase from '../../firebase';
import { Link } from 'react-router-dom';

class ForgotPassword extends Component {
  constructor() {
    super();
    this.state = {
      email: '',
      status: null
    };
  }

  handleChange = (evt) => {
    const {name, value} = evt.target;
    this.setState({[name]: value});
  }

  handleSubmit = () => {
    firebase.auth().sendPasswordResetEmail(this.state.email);
    const msg = `ğŸš€ A link was sent to ${this.state.email} - Check your email shortly!ğŸš€ `;
    this.setState({status: msg});
  }


  render() {
    return (
      <div className="ForgotPassword">
        <h1>Reset Password</h1>
        <input type="text" name="email" placeholder="Email" value={this.state.email} onChange={this.handleChange}/>
        <span>{this.state.status}</span>
        <button onClick={this.handleSubmit}>Send Reset Link</button>
        <Link to="/signin">Back to Sign In</Link>
      </div>
    )
  }
}
export default ForgotPassword;