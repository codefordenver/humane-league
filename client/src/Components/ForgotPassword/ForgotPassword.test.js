import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import ForgotPassword from './ForgotPassword';

describe('ForgotPassword component tests', () => {
  let renderedForgotPassword;
  let defaultState; 
  
  beforeEach(() => {
    renderedForgotPassword = shallow( <ForgotPassword /> );
    defaultState ={
      email: '',
      status: null
    }
  })
  it('renders without crashing', () => {
    expect(renderedForgotPassword).toBeDefined();
    expect(renderedForgotPassword.state()).toEqual(defaultState);
  });
  it('matches SnapShot', () => {
    expect(renderedForgotPassword).toMatchSnapshot();
  })
})