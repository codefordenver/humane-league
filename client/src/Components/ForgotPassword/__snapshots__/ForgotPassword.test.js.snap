// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPassword component tests matches SnapShot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ForgotPassword />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Reset Password
</h1>,
        <input
          name="email"
          onChange={[Function]}
          placeholder="Email"
          type="text"
          value=""
/>,
        <span />,
        <button
          onClick={[Function]}
>
          Send Reset Link
</button>,
        <Link
          replace={false}
          to="/signin"
>
          Back to Sign In
</Link>,
      ],
      "className": "ForgotPassword",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Reset Password",
        },
        "ref": null,
        "rendered": "Reset Password",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "email",
          "onChange": [Function],
          "placeholder": "Email",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": null,
        },
        "ref": null,
        "rendered": null,
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Send Reset Link",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Send Reset Link",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Back to Sign In",
          "replace": false,
          "to": "/signin",
        },
        "ref": null,
        "rendered": "Back to Sign In",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Reset Password
</h1>,
          <input
            name="email"
            onChange={[Function]}
            placeholder="Email"
            type="text"
            value=""
/>,
          <span />,
          <button
            onClick={[Function]}
>
            Send Reset Link
</button>,
          <Link
            replace={false}
            to="/signin"
>
            Back to Sign In
</Link>,
        ],
        "className": "ForgotPassword",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Reset Password",
          },
          "ref": null,
          "rendered": "Reset Password",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "email",
            "onChange": [Function],
            "placeholder": "Email",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": null,
          },
          "ref": null,
          "rendered": null,
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Send Reset Link",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Send Reset Link",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Back to Sign In",
            "replace": false,
            "to": "/signin",
          },
          "ref": null,
          "rendered": "Back to Sign In",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
